Battle.js

```diff
@@ -0,0 +1,124 @@
+/*TODO: 
+Main battle screen, should display 2 knights with health bars and a 
+button for each different ability*/
+
+import { StyleSheet, Text, View, Button, Image, Dimensions, Pressable } from 'react-native';
+import React, { useState, useRef, useEffect } from 'react';
+import { HealthBar } from "./HealthBar.js"
+import { score } from "./Dodge.js"
+import { useFocusEffect } from '@react-navigation/native';
+import { hookstate, useHookstate, createState } from '@hookstate/core';
+// .value: Gets the value of the state
+// .set(): Sets the value of the state
+
+
+const MAXwidth = Dimensions.get('window').width;
+const MAXheight = Dimensions.get('window').height;
+//const [health, setHealth] = useState(100);
+const health2 = createState(100);
+
+export default function Battle({ navigation, route }) {
+    const healthState2 = useHookstate(health2);
+    //function which runs when the screen is 'focused' e.g. visible
+    useFocusEffect(
```

Comment:
This is also about Hooks (see https://reactjs.org/docs/hooks-effect.html), and probably isn't really the right tool to use here. This code actually says as long as this navigation tab is focused "listen for any changes in <empty list of things that could change on line 26>". So maybe it *happens* to be called once when the screen gets focus, but its probably not quite what you want.

If you want something to happen when the Battle Screen gains focus, I would do it like this:
```
export default function Battle() { // or inside Battle's constructor() { ... }
  React.useEffect(() => { // Tell react to start listening for a kind of update
      return navigation.addListener('focus', () => {
        // Do something here.
      });
    }, [navigation]); // updates will trigger whenever there is a state change to navigation
}
```

But you could also have an explicit `miniGameFinished()` function (probably a method as part of the Battle class) which you pass to the minigame. When the minigame is finished, it can both navigate to the appropriate screen *and* call `miniGameFinished(<minigame results>)` which can update the battle screen accordingly (i.e. it would call `battle.setState({health: battle.state.health - damage})` or something similar).